src/actions/hover.rs

```diff
@@ -782,15 +782,19 @@ fn format_object(rustfmt: Rustfmt, fmt_config: &FmtConfig, the_type: String) ->
         format!("{}{{}}", trimmed)
     };
 
-    let formatted = match rustfmt.format(object.clone(), config) {
-        Ok(lines) => match lines.rfind('{') {
+    let formatted = match std::panic::catch_unwind(|| rustfmt.format(object.clone(), config)) {
```

Comment:
This seems too ad-hoc - not sure if that's specifically only to pin down the panic mentioned. Could we `catch_unwind` [here](https://github.com/rust-lang/rls/blob/master/src/actions/format.rs#L94) instead?

Reply:
Only the actual panic that occurred, it's a workaround to the `format` bug that allows a panic at all. It should return an `Err` surely.

Maybe I'll add test reproducing the panic. This way we can remove the catch is this gets fixed upstream.

Reply:
Seems that we can’t catch this panic, see https://github.com/rust-lang/rustfmt/issues/3239

Reply:
`catch_unwind` absolutely did work here though and allowed the test harness to run properly. I wouldn't have added it otherwise.

Reply:
You're right, my bad - we resume unwinding on fatal error without the backtrace instead of aborting. The panic we don't catch however can also stem from [creating parser](https://github.com/rust-lang/rustfmt/issues/3239#issuecomment-445489769) as well so I think we can fix that on Rustfmt side.

Reply:
@alexheretic could we remove the catching panic logic? This shouldn't surely panic (https://github.com/rust-lang/rustfmt/pull/3240) and should be fixed when the PR upstream lands.